openstack/compute/v2/servers/results.go

```diff
@@ -82,6 +84,47 @@ type CreateImageResult struct {
 	gophercloud.Result
 }
 
+// GetPasswordResult represent the result of a get os-server-password operation.
+type GetPasswordResult struct {
+	gophercloud.Result
+}
+
+// ExtractPassword gets the encrypted password.
+// If privateKey != nil the password is decrypted with the private key.
+// If privateKey == nil the encrypted password is returned and can be decrypted with:
+//   echo '<pwd>' | base64 -D | openssl rsautl -decrypt -inkey <private_key>
+func (r GetPasswordResult) ExtractPassword(privateKey *rsa.PrivateKey) (string, error) {
```

Comment:
Is an OpenStack password guaranteed to be of type RSA? If not, we should probably have `privateKey` be of type `crypto.PrivateKey`. Then down in `decryptPassword` we can `switch` on the type of `privateKey` to figure out the correct decryption scheme.